File: csharp/problems/lc13.cs
using System;
using System.Collections.Generic;

namespace lc13
{



    public class Solution
    {
        public int RomanToInt(string s)
        {
            Dictionary<char, int> m = new Dictionary<char, int>();
            m.Add('I', 1);
            m.Add('V', 5);
            m.Add('X', 10);
            m.Add('L', 50);
            m.Add('C', 100);
            m.Add('D', 500);
            m.Add('M', 1000);



            int ans = 0;

            int n = s.Length;
            int i = 0;

            while (i < n - 1)
            {

                int sign = 1;

                if (m[s[i]] < m[s[i + 1]])
                {
                    sign = -1;
                }

                ans += (sign * m[s[i]]);
                i++;

            }

            ans += m[s[i]];



            return ans;

        }

        public void Test()
        {
            // lc151.Solution s = new lc151.Solution();
            string s = "MCMXCIV";
            int ans = RomanToInt(s);
            Console.WriteLine($"ans is {ans}");
        }


    }

}

File: python/problems/lc13.py
from pprint import pprint
from typing import List, Deque
from collections import deque


romans = {
    "I": (0, 1),
    "V": (1, 5),
    "X": (2, 10),
    "L": (3, 50),
    "C": (4, 100),
    "D": (5, 500),
    "M": (6, 1000),
}


class Solution:
    def romanToInt(self, s: str) -> int:

        ans = 0


        for i in range(len(s) -1):
            sign = 1 if romans[s[i]][0] >= romans[s[i+1]][0] else -1

            ans += (sign * romans[s[i]][1])

        return ans + romans[s[-1]][1]


def main():
    s = "MCMXCIV"
    # s = "LVIII"

    solution = Solution()
    ans = solution.romanToInt(s)
    print("ans is ----------------------------------------")
    pprint(ans)


if __name__ == "__main__":
    main()
